languages/elixir/exercises/concept/basic-strings/.meta/design.md

```diff
@@ -0,0 +1,51 @@
+# Design
+
+## Goal
+
+The goal of this exercise is to teach the student how strings are implemented and used in Elixir.
+
+## Learning objectives
+
+After completing this exercise, the student should:
+
+- Know how to concatenate strings.
+- Know how to manipulate strings.
+- Know how to perform string interpolation.
+- Know how to create multiline strings (heredocs).
+
+## Out of scope
+
+- Strings as binaries for binary pattern matching.
+- Regular expressions.
+- Sigils.
+- Charlists.
+
+## Prerequisites
+
+This exercise's prerequisites are:
+
+- `lists`: needs to be able to work with the return value of `String.split`.
+- `basic-pattern-matching`: needs to be able to work with the return value of `String.split`.
+
+## Concepts
+
+The concepts this exercise unlocks are:
+
+- `basic-strings`: knows how to concatenate strings and modify them using the _String module_.
+- `string-interpolation`: knows how to interpolate expressions inside of strings.
+- `heredocs`: knows how to create multiline strings using the heredoc syntax.
+
+## Representer
+
+This exercise does not require any specific representation logic to be added to the [representer][representer].
+
+## Analyzer
+
+This exercise could benefit from the following rules added to the the [analyzer][analyzer]:
+
+- Verify that the `initial/1` function calls the `first_letter/1` function.
+- Verify that the `initials/1` function calls the `initial/1` function.
+- Verify that the `pair/1` function calls the `initials/1` function and uses `"""` to create the string.
```

Comment:
I confess that I have no idea what the Analyzer is, but I looked at other Elixir exercises' `design.md` files and made some assumptions about what I can write here.

Reply:
the analyzer is the static code analysis component which we can eventually add in to all of this.  So these are great suggestions for that.  Referencing your other comment, it would also allow us to check for pattern matching rather than hd(tl(split_names))

Reply:
:point_up: An analyzer's task is to automatically comment on code (without human interaction required). It should be used when the tests cannot verify certain aspects of the code that you do want to comment on. Does that make sense?

Reply:
> Does that make sense?

Yes, thank you!

> it would also allow us to check for pattern matching rather than hd(tl(split_names))

Since this is not crucial for learning about strings, I would not really care if somebody uses that as a workaround for pattern-matching a list.